File: garbage_system/simulation.py
import random

def generate_normal(mean, sd):
    # Box-Muller transform approximation using Central Limit Theorem
    sum_vals = sum(random.random() for _ in range(12))
    return mean + sd * (sum_vals - 6)

def single_queue_simulation(run=10):
    mean_iat = 10.0
    sd_iat = 1.5
    mean_st = 9.5
    sd_st = 1.0

    cat = 0.0  # Cumulative Arrival Time
    se = 0.0   # Service End Time
    cwt = 0.0  # Cumulative Waiting Time
    cit = 0.0  # Cumulative Idle Time

    # Print header with proper column width for alignment
    print(f"{'i':<3}{'IAT':<10}{'CAT':<10}{'SB':<10}{'ST':<10}{'SE':<10}{'WT':<10}{'IT':<10}")

    for j in range(1, run + 1):
        iat = generate_normal(mean_iat, sd_iat)
        cat += iat

        if cat <= se:
            sb = se
            wt = se - cat
            cwt += wt
            it = 0.0
        else:
            sb = cat
            wt = 0.0
            it = sb - se
            cit += it

        st = generate_normal(mean_st, sd_st)
        se = sb + st

        # Print each line of the table with consistent column width
        print(f"{j:<3}{iat:<10.3f}{cat:<10.3f}{sb:<10.3f}{st:<10.3f}{se:<10.3f}{wt:<10.3f}{it:<10.3f}")

    # Calculate average waiting time and percentage capacity utilization
    awt = cwt / run
    pcu = ((cat - cit) * 100) / cat

    # Print final results
    print(f"\n{'Average Waiting Time:':<20}{awt:.3f}")
    print(f"{'Percentage Capacity Utilization:':<20}{pcu:.2f}%")

# Run the simulation
single_queue_simulation()

File: garbage_system/simulation.php
<?php
// Queue Simulation in PHP

$run = 10;
$mean = 10.0;
$sd = 1.5;
$mue = 9.5;
$sigma = 1.0;

$wt = 0.0;
$cit = 0.0;
$cat = 0.0;
$se = 0.0;
$cwt = 0.0;

echo str_pad("i", 5) .
     str_pad("IAT", 10) .
     str_pad("CAT", 10) .
     str_pad("SB", 10) .
     str_pad("ST", 10) .
     str_pad("SE", 10) .
     str_pad("WT", 10) .
     str_pad("IT", 10) . PHP_EOL;

for ($j = 1; $j <= $run; $j++) {
    // Generate inter-arrival time
    $sum = 0.0;
    for ($i = 0; $i < 12; $i++) {
        $sum += mt_rand() / mt_getrandmax();
    }
    $x1 = $mean + $sd * ($sum - 6.0);
    $iat = $x1;
    $cat += $iat;

    // Determine service begin time and waiting time
    if ($cat <= $se) {
        $sb = $se;
        $wt = $se - $cat;
        $cwt += $wt;
        $it = 0;
    } else {
        $sb = $cat;
        $it = $sb - $se;
        $cit += $it;
        $wt = 0;
    }

    // Generate service time
    $sum = 0.0;
    for ($i = 0; $i < 12; $i++) {
        $sum += mt_rand() / mt_getrandmax();
    }
    $x2 = $mue + $sigma * ($sum - 6.0);
    $st = $x2;
    $se = $sb + $st;

    // Output row
    echo str_pad($j, 5) .
         str_pad(number_format($iat, 2), 10) .
         str_pad(number_format($cat, 2), 10) .
         str_pad(number_format($sb, 2), 10) .
         str_pad(number_format($st, 2), 10) .
         str_pad(number_format($se, 2), 10) .
         str_pad(number_format($wt, 2), 10) .
         str_pad(number_format($it, 2), 10) . PHP_EOL;
}

$awt = $cwt / $run;
$pcu = (($cat - $cit) * 100.0) / $cat;

echo "\nAverage Waiting Time: " . number_format($awt, 2) . " minutes\n";
echo "Percentage Capacity Utilization: " . number_format($pcu, 2) . "%\n";
?>
